transaction/std/src/memo_builder/gift_code_sender_memo_builder.rs

```diff
@@ -154,26 +183,61 @@ mod tests {
 
         // Build the memo payload and get the data
         let amount = Amount::new(42, 0.into());
+        let fee = Amount::new(10, 0.into());
+        builder.set_fee(fee).unwrap();
+
         let memo_payload = build_change_memo_with_amount(&mut builder, amount).unwrap();
 
         // Verify memo data can be verified after writing the first change memo
-        let derived_note = GiftCodeSenderMemo::from(memo_payload.get_memo_data());
-        assert_eq!(note, derived_note.sender_note().unwrap());
+        let derived_memo = GiftCodeSenderMemo::from(memo_payload.get_memo_data());
+        assert_eq!(note, derived_memo.sender_note().unwrap());
+        assert_eq!(fee.value, derived_memo.get_fee());
 
         // Verify attempting to write two change memos fail
         let memo_payload_2 = build_change_memo_with_amount(&mut builder, amount);
-        assert!(matches!(
-            memo_payload_2,
-            Err(NewMemoError::MultipleChangeOutputs)
-        ));
+        assert_eq!(memo_payload_2, Err(NewMemoError::MultipleChangeOutputs));
     }
 
     #[test]
     fn test_gift_code_sender_note_builder_creation_fails_with_invalid_note() {
         // Create Memo Builder with an input longer than allowed
-        let note_bytes = [b'6'; GiftCodeSenderMemo::MEMO_DATA_LEN + 1];
+        let note_bytes = [b'6'; GiftCodeSenderMemo::NOTE_DATA_LEN + 1];
         let note = std::str::from_utf8(&note_bytes).unwrap();
         let builder = GiftCodeSenderMemoBuilder::new(note);
-        assert!(matches!(builder, Err(NewMemoError::BadInputs(_))));
+        assert_matches!(builder, Err(NewMemoError::BadInputs(_)));
+    }
+
+    #[test]
+    fn test_gift_code_sender_memo_builder_fee_token_cannot_be_different_from_change_token() {
+        let note = "It's MEMO TIME!!";
+        let mut builder = GiftCodeSenderMemoBuilder::new(note).unwrap();
+        let amount = Amount::new(42, 0.into());
+        let fee = Amount::new(1, 9001.into());
+
+        // Set a fee with a different token id
+        builder.set_fee(fee).unwrap();
+
+        // Attempt to build the memo
+        let memo_payload = build_change_memo_with_amount(&mut builder, amount);
+
+        // Ensure memo creation fails
+        assert_eq!(memo_payload, Err(NewMemoError::MixedTokenIds))
+    }
+
+    #[test]
+    fn test_gift_code_sender_memo_builder_set_fee_fails_when_exceeding_max_fee() {
+        let note = "It's MEMO TIME!!";
+        let mut builder = GiftCodeSenderMemoBuilder::new(note).unwrap();
+        let fee = Amount::new(u64::MAX, 0.into());
```

Comment:
This should be testing boundary conditions:
* `GiftCodeSenderMemo::MAX_FEE`
* `GiftCodeSenderMemo::MAX_FEE + 1`

Reply:
Addressed!